experimental/packages/opentelemetry-sdk-node/src/TracerProviderWithEnvExporter.ts

```diff
@@ -0,0 +1,109 @@
+import { OTLPTraceExporter as OTLPGrpcTraceExporter} from '@opentelemetry/exporter-trace-otlp-grpc';
+import { ZipkinExporter } from '@opentelemetry/exporter-zipkin';
+import { JaegerExporter } from '@opentelemetry/exporter-jaeger';
+
+export class TracerProviderWithEnvExporters extends NodeTracerProvider {
+  public configuredExporters: SpanExporter[] = [];
+  public spanProcessors: (BatchSpanProcessor | SimpleSpanProcessor)[] | undefined;
+
+  static configureOtlp(): SpanExporter {
+    const protocol = this.getOtlpProtocol();
+
+    switch (protocol) {
+      case 'grpc':
+        return new OTLPGrpcTraceExporter;
+      case 'http/json':
+        return new OTLPHttpTraceExporter;
+      case 'http/protobuf':
+        return new OTLPProtoTraceExporter;
+      default:
+        diag.warn(`Unsupported OTLP traces protocol: ${protocol}. Using http/protobuf.`);
+        return new OTLPProtoTraceExporter;
+    }
+  }
+
+  static getOtlpProtocol(): string {
+    const parsedEnvValues = getEnvWithoutDefaults();
+
+    return parsedEnvValues.OTEL_EXPORTER_OTLP_TRACES_PROTOCOL ??
+          parsedEnvValues.OTEL_EXPORTER_OTLP_PROTOCOL ??
+          getEnv().OTEL_EXPORTER_OTLP_TRACES_PROTOCOL ??
+          getEnv().OTEL_EXPORTER_OTLP_PROTOCOL;
+  }
+
+  protected static override _registeredExporters = new Map<
+    string,
+    () => SpanExporter
+      >([
+        ['otlp', () => this.configureOtlp()],
+        ['zipkin', () => new ZipkinExporter],
+        ['jaeger', () => new JaegerExporter],
+        ['console', () => new ConsoleSpanExporter]
+      ]);
+
+  public constructor(config: NodeTracerConfig = {}) {
+    super(config);
+    let traceExportersList = this.filterBlanksAndNulls(Array.from(new Set(getEnv().OTEL_TRACES_EXPORTER.split(','))));
+
+    if (traceExportersList.length === 0 || traceExportersList[0] === 'none') {
+      diag.warn('OTEL_TRACES_EXPORTER contains "none" or is empty. SDK will not be initialized.');
+    } else {
+      if (traceExportersList.length > 1 && traceExportersList.includes('none')) {
+        diag.warn('OTEL_TRACES_EXPORTER contains "none" along with other exporters. Using default otlp exporter.');
+        traceExportersList = ['otlp'];
+      }
+
+      traceExportersList.forEach(exporterName => {
+        const exporter = this._getSpanExporter(exporterName);
+        if (exporter) {
+          this.configuredExporters.push(exporter);
+        } else {
+          diag.warn(`Unrecognized OTEL_TRACES_EXPORTER value: ${exporterName}.`);
+        }
+      });
+
+      if (this.configuredExporters.length > 0) {
+        this.spanProcessors = this.configureSpanProcessors(this.configuredExporters);
+      } else {
+        diag.warn('Unable to set up trace exporter(s) due to invalid exporter and/or protocol values.');
+      }
+    }
+  }
+
+  public configureSpanProcessors(exporters: SpanExporter[]): (BatchSpanProcessor | SimpleSpanProcessor)[] {
```

Comment:
Yes I can make it private. I would just have to [remove the spy for this method](https://github.com/open-telemetry/opentelemetry-js/blob/e02d73981813f0efec72642e7e3d4ee10b53d0c6/experimental/packages/opentelemetry-sdk-node/test/sdk.test.ts#L490) and remove this method from tests. I can do the same for the `filterBlanksAndNulls` method as well since I made it public only for testing purposes.

Reply:
private methods can be tested by using the bracket syntax `someObject['_somePrivateProperty']` or casting to `any` but I would argue that most of the time private methods shouldn't be explicitly tested anyway. The public interface is the most important part to keep stable.

Reply:
Got it. I've updated this so that these methods are private but I've also added some tests to test `TracerProviderWithEnvExporters` specifically.